### PR TITLE
Add demo code for graph vis class

### PR DESCRIPTION
**Pull request by javidhsueh**
Add a new folder 'graph' under src/demos.
There are 5 steps to complete the interactive graph visualization application.

The starting code for each step is separated into different folders:
- 0-starting-code: only the basic scaffold. Users will add GraphRender component.
- 1-basic-graph: the result from the step 0. Users will complete the app.js here.
- 2-graph-layout: the result from the step 1. Users will add the layout-engine.
- 3-interactive-graph: the result from the step 2. Users will add onHover interaction.
- 4-final-version: the complete code of the interactive graph visualization application.

### CODE REVIEW

File: src/demos/graph/3-interactive-graph/src/graph-render.js [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+import DeckGL, {
+  LineLayer,
+  ScatterplotLayer,
+  OrthographicViewport,
+  COORDINATE_SYSTEM
+} from 'deck.gl';
+
+export default class GraphRender extends PureComponent {
+
+  creatViewport() {
```
Comment (jckr) [inline src/demos/graph/3-interactive-graph/src/graph-render.js]: same

File: src/demos/graph/4-final-version/src/graph-render.js [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+import DeckGL, {
+  LineLayer,
+  ScatterplotLayer,
+  OrthographicViewport,
+  COORDINATE_SYSTEM
+} from 'deck.gl';
+
+export default class GraphRender extends PureComponent {
+
+  creatViewport() {
```
Comment (jckr) [inline src/demos/graph/4-final-version/src/graph-render.js]: ditto

File: src/demos/graph/1-basic-graph/src/graph-render.js [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+  LineLayer,
+  ScatterplotLayer,
+  OrthographicViewport,
+  COORDINATE_SYSTEM
+} from 'deck.gl';
+
+export default class GraphRender extends PureComponent {
+
+  creatViewport() {
+    const {height, width} = this.props;
```
[the remark this answers is not in the document]
Comment (javidhsueh) [reply src/demos/graph/1-basic-graph/src/graph-render.js]: will do

File: src/demos/graph/1-basic-graph/src/graph-render.js [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+import DeckGL, {
+  LineLayer,
+  ScatterplotLayer,
+  OrthographicViewport,
+  COORDINATE_SYSTEM
+} from 'deck.gl';
+
+export default class GraphRender extends PureComponent {
+
+  creatViewport() {
```
[the remark this answers is not in the document]
Comment (javidhsueh) [reply src/demos/graph/1-basic-graph/src/graph-render.js]: typo, will fix it

File: src/demos/graph/2-graph-layout/src/graph-render.js [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+import DeckGL, {
+  LineLayer,
+  ScatterplotLayer,
+  OrthographicViewport,
+  COORDINATE_SYSTEM
+} from 'deck.gl';
+
+export default class GraphRender extends PureComponent {
+
+  creatViewport() {
```
[the remark this answers is not in the document]
Comment (javidhsueh) [reply src/demos/graph/2-graph-layout/src/graph-render.js]: typo, will fix it

File: src/demos/graph/3-interactive-graph/src/graph-render.js (hunk 1 above)
Comment (javidhsueh) [reply src/demos/graph/3-interactive-graph/src/graph-render.js]: typo, will fix it

File: src/demos/graph/4-final-version/src/graph-render.js (hunk 2 above)
Comment (javidhsueh) [reply src/demos/graph/4-final-version/src/graph-render.js]: typo, will fix it

File: src/demos/graph/2-graph-layout/src/app.js [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+    this.setState({
+      viewport: {
+        width: window.innerWidth,
+        height: window.innerHeight
+      }
+    });
+  }
+
+  processData() {
+    if (sampleGraph) {
```
[the remark this answers is not in the document]
Comment (javidhsueh) [reply src/demos/graph/2-graph-layout/src/app.js]: We have the same check in the original tutorial?
https://uber-common.github.io/vis-academy/#/building-a-geospatial-app/2-map-data-overlays-scatterplot

[the remark this answers is not in the document]
Comment (jckr) [reply src/demos/graph/2-graph-layout/src/app.js]: that's fair. that's a remnant of the time when we were actually getting that data through an XHR query

File: src/demos/graph/1-basic-graph/src/app.js [hunk 7]
```diff
@@ -0,0 +1,19 @@
+/* global window */
```
[the remark this answers is not in the document]
Comment (javidhsueh) [reply src/demos/graph/1-basic-graph/src/app.js]: yes

File: src/demos/graph/2-graph-layout/src/app.js [hunk 8]
```diff
@@ -0,0 +1,105 @@
...
+
+export default class App extends Component {
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      viewport: {
+        width: window.innerWidth,
+        height: window.innerHeight
+      },
```
[the remark this answers is not in the document]
Comment (javidhsueh) [reply src/demos/graph/2-graph-layout/src/app.js]: I don't call this.onResize on componentDidMount.
Also, this is to follow our previous example too:
https://github.com/uber-common/vis-academy/blob/master/src/demos/building-a-geospatial-app/1-starting-with-map/app.js

**Pull request merged**